packages/ui/src/Data/Virtualizations/VirtListItem.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  Button,
+  DropdownKebab,
+  ListViewIcon,
+  ListViewItem,
+  MenuItem,
+  OverlayTrigger,
+  Tooltip,
+} from 'patternfly-react';
+import * as React from 'react';
+
+export interface IVirtListItemProps {
+  virtName: string;
+  virtDescription: string;
+  i18nDraft: string;
+  i18nDraftTip: string;
+  icon?: string;
+  i18nEdit: string;
+  i18nEditTip?: string;
+  i18nPublished: string;
+  i18nPublishedTip: string;
+  i18nDelete: string;
+  i18nExport: string;
+  i18nUnpublish: string;
+  i18nPublish: string;
+  isPublished: boolean;
+  onDelete: (virtName: string) => void;
+  onEdit: (virtName: string) => void;
+  onExport: (virtName: string) => void;
+  onUnpublish: (virtName: string) => void;
+  onPublish: (virtName: string) => void;
+}
+
+export class VirtListItem extends React.Component<IVirtListItemProps> {
+  public constructor(props: IVirtListItemProps) {
+    super(props);
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleExport = this.handleExport.bind(this);
+    this.handleUnpublish = this.handleUnpublish.bind(this);
+    this.handlePublish = this.handlePublish.bind(this);
+  }
+  public getEditTooltip() {
+    return (
+      <Tooltip id="detailsTip">
+        {this.props.i18nEditTip ? this.props.i18nEditTip : this.props.i18nEdit}
+      </Tooltip>
+    );
+  }
+
+  public getPublishedTooltip() {
+    return (
+      <Tooltip id="detailsTip">
+        {this.props.isPublished
+          ? this.props.i18nPublishedTip
+          : this.props.i18nDraftTip}
+      </Tooltip>
+    );
+  }
+
+  public handleDelete() {
```

Comment:
recommend being consistent with the handleXYZ functions.  If you use the arrow notation like in handleEdit, I think you can do away with the constructor - you wont need the binds

Reply:
Agree, although I'd suggest using the bind version. [I commented about this in the past.](https://github.com/syndesisio/syndesis-react-poc/pull/56#discussion_r248996659)